packages/abi/src/gen/renderers/ts/typers/struct.ts

```diff
@@ -0,0 +1,158 @@
+import { assertUnreachable } from '@fuel-ts/utils';
+
+import { ENUM_REGEX, STRUCT_REGEX, TUPLE_REGEX } from '../../../../matchers/sway-type-matchers';
+import type { AbiConcreteType, AbiTypeComponent, AbiMetadataType } from '../../../../parser';
+
+import { flattenImports } from './helpers';
+import type { TyperReturn, Typer, GlobalTyper, TyperAbiType } from './types';
+
+function wrapStructContent(text: string, wrap: '{}' | '[]' | 'Enum'): string {
+  switch (wrap) {
+    case '{}':
+      return `{ ${text} }`;
+    case '[]':
+      return `[${text}]`;
+    case 'Enum': {
+      const wrappedAsObj = wrapStructContent(text, '{}');
+      return `Enum<${wrappedAsObj}>`;
+    }
+    default:
+      return assertUnreachable(wrap);
+  }
+}
+
+function componentMapper(
+  c: AbiTypeComponent | { name: string; type: AbiConcreteType | AbiMetadataType },
+  includeName: boolean,
+  generateTsType: GlobalTyper
+): TyperReturn {
+  const mapped = generateTsType({ abiType: c.type, asReference: true });
+
+  if (!includeName) {
+    return mapped;
+  }
+
+  return {
+    ...mapped,
+    input: `${c.name}: ${mapped.input}`,
+    output: `${c.name}: ${mapped.output}`,
+  };
+}
+
+export function mapComponents(params: {
```

Comment:
This function seems like it's used by a lot of different _typers_ and is not specific to the struct. Could this be a in the helpers? Adding comments to explain what the method does would be beneficial.